pkg/nsx/services/ipaddressallocation/ipaddressallocation.go

```diff
@@ -180,6 +183,67 @@ func (service *IPAddressAllocationService) DeleteIPAddressAllocation(obj interfa
 	return nil
 }
 
+func (service *IPAddressAllocationService) DeleteIPAddressAllocationByNamespacedName(namespace, name string) error {
+	// First, we need to find the IPAddressAllocation in our store
+	allIPAddressAllocations := service.ipAddressAllocationStore.List()
+	var targetIPAddressAllocation *model.VpcIpAddressAllocation
+
+	for _, obj := range allIPAddressAllocations {
+		ipAddressAllocation, ok := obj.(*model.VpcIpAddressAllocation)
+		if !ok {
+			continue
+		}
+
+		for _, tag := range ipAddressAllocation.Tags {
+			if *tag.Scope == common.TagScopeNamespace && *tag.Tag == namespace {
+				targetIPAddressAllocation = ipAddressAllocation
+				break
+			}
+		}
```

Comment:
Get all ipAddressAllocation with this namespace? As when the new one is created and the old one is not deleted there might be 2 ipAddressAllocation associated with the same namespace

Reply:
ok